pyanaconda/payload/dnf/payload.py

```diff
@@ -139,9 +140,19 @@ def proxy(self):
         """
         return self._dnf_proxy
 
+    def get_source_proxy(self):
+        """Get the DBus proxy of the RPM source.
+
+        :return: a DBus proxy
+        """
+        return get_source(self.proxy, SOURCE_TYPE_REPO_FILES)
+
     @property
     def is_hmc_enabled(self):
-        return self.data.method.method == "hmc"
+        # FIXME: Remove this property and check the type directly.
+        source_proxy = self.get_source_proxy()
+        source_type = source_proxy.Type
+        return source_type == SOURCE_TYPE_HMC
```

Comment:
I would merge these two lines. I don't see reason here to have it on two lines.

Reply:
Fixed.